01_debian-patch by Junichi Uekawa <dancer@debian.org>
Debian modifications to ladspa-sdk
Index: ladspa-sdk/src/makefile
===================================================================
--- ladspa-sdk.orig/src/makefile	2009-06-05 13:50:20.000000000 +1000
+++ ladspa-sdk/src/makefile	2009-06-05 13:54:46.000000000 +1000
@@ -4,9 +4,9 @@
 #
 # Change these if you want to install somewhere else.
 
-INSTALL_PLUGINS_DIR	=	/usr/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/bin/
+INSTALL_PLUGINS_DIR	?=	/usr/lib/ladspa/
+INSTALL_INCLUDE_DIR	?=	/usr/include/
+INSTALL_BINARY_DIR	?=	/usr/bin/
 
 ###############################################################################
 #
@@ -15,7 +15,7 @@
 
 INCLUDES	=	-I.
 LIBRARIES	=	-ldl -lm
-CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
+CFLAGS		=	$(INCLUDES) -O3 -fPIC
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
 			../plugins/delay.so				\
@@ -35,11 +35,11 @@
 
 ../plugins/%.so:	plugins/%.c ladspa.h
 	$(CC) $(CFLAGS) -o plugins/$*.o -c plugins/$*.c
-	$(LD) -o ../plugins/$*.so plugins/$*.o -shared
+	$(LD) -lc $(LIBRARIES) -o ../plugins/$*.so plugins/$*.o -shared
 
 ../plugins/%.so:	plugins/%.cpp ladspa.h
 	$(CPP) $(CXXFLAGS) -o plugins/$*.o -c plugins/$*.cpp
-	$(CPP) -o ../plugins/$*.so plugins/$*.o -shared
+	$(CPP) $(LIBRARIES) -o ../plugins/$*.so plugins/$*.o -shared
 
 ###############################################################################
 #
@@ -59,9 +59,9 @@
 	@echo Test complete.
 
 install:	targets
-	-mkdirhier $(INSTALL_PLUGINS_DIR)
-	-mkdirhier $(INSTALL_INCLUDE_DIR)
-	-mkdirhier $(INSTALL_BINARY_DIR)
+	-mkdir -p $(INSTALL_PLUGINS_DIR)
+	-mkdir -p $(INSTALL_INCLUDE_DIR)
+	-mkdir -p $(INSTALL_BINARY_DIR)
 	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
 	cp ladspa.h $(INSTALL_INCLUDE_DIR)
 	cp ../bin/* $(INSTALL_BINARY_DIR)
Index: ladspa-sdk/src/search.c
===================================================================
--- ladspa-sdk.orig/src/search.c	2009-06-05 13:51:31.000000000 +1000
+++ ladspa-sdk/src/search.c	2009-06-05 13:53:47.000000000 +1000
@@ -99,10 +99,10 @@
 
   pcLADSPAPath = getenv("LADSPA_PATH");
   if (!pcLADSPAPath) {
+    pcLADSPAPath="/usr/lib/ladspa/";
     fprintf(stderr,
 	    "Warning: You do not have a LADSPA_PATH "
 	    "environment variable set.\n");
-    return;
   }
   
   pcStart = pcLADSPAPath;
Index: ladspa-sdk/src/load.c
===================================================================
--- ladspa-sdk.orig/src/load.c	2009-06-05 13:51:31.000000000 +1000
+++ ladspa-sdk/src/load.c	2009-06-05 13:53:47.000000000 +1000
@@ -54,6 +54,8 @@
        to search. */
 
     pcLADSPAPath = getenv("LADSPA_PATH");
+    if (!pcLADSPAPath)
+      pcLADSPAPath="/usr/lib/ladspa";
     
     if (pcLADSPAPath) {
 
